cpp/collections_module/algorithm/algorithm.cpp

```diff
@@ -0,0 +1,25 @@
+#include "algorithm.hpp"
+#include <unordered_set>
+
+void Collections::toSet(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *result, mgp_memory *memory) {
+  mgp::memory = memory;
+  auto arguments = mgp::List(args);
+  const auto record_factory = mgp::RecordFactory(result);
+  try {
+    mgp::List list = arguments[0].ValueList();
+    std::unordered_set<mgp::Value, std::hash<mgp::Value>> set;
```

Comment:
You can construct this set from a list directly, like this
std::unordered_set<..> set{list.begin(), list.end())
https://en.cppreference.com/w/cpp/container/unordered_set/unordered_set, constructor (6)